.gitlab-ci.yml

```diff
@@ -283,6 +283,9 @@ gcp-stg:
     - rake destroy_module[locust] || true
     # Remove all SA keys except current one to prevent hitting 10 keys per SA limit (GPII-3299)
     - rake destroy_sa_keys || true
+    # Rotate Terraform state key and secrets KMS key
+    - rake rotate_tfstate_key # We want CI to stop in case any of these after scripts fail
```

Comment:
I don't think this belongs in after_script. This doesn't seem like something we want to run regardless of whether the build was successful -- it's not a "cleanup" step.

Furthermore, I think this should happen earlier in the Job, right after `configure_serviceaccount_ci_restore`. That way, if rotating tfstate or secrets breaks something, we'll find out in the build where the failure occurred instead of in the next build.

Reply:
Right, I am not thinking clearly on Wednesdays!